From 827d6816fb18cec2e3b2271e3e9b21ca964c0136 Mon Sep 17 00:00:00 2001
From: b02247 <b02247@freescale.com>
Date: Thu, 28 Feb 2013 15:18:15 +0800
Subject: [PATCH 08/25] ENGR00242652 user can't tune the volume on hdmidongle

specify hdmidongle when tuning the volume

Signed-off-by: b02247 <b02247@freescale.com>

Conflicts:
	audio/AudioPolicyManagerBase.cpp
---
 audio/AudioPolicyManagerBase.cpp                 | 8 +++++++-
 include/hardware_legacy/AudioPolicyManagerBase.h | 2 ++
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/audio/AudioPolicyManagerBase.cpp b/audio/AudioPolicyManagerBase.cpp
index 0f2cce6..508e44e 100644
--- a/audio/AudioPolicyManagerBase.cpp
+++ b/audio/AudioPolicyManagerBase.cpp
@@ -31,6 +31,8 @@
 // A device mask for all audio output devices that are considered "remote" when evaluating
 // active output devices in isStreamActiveRemotely()
 #define APM_AUDIO_OUT_DEVICE_REMOTE_ALL  AUDIO_DEVICE_OUT_REMOTE_SUBMIX
+#define PRODUCT_DEVICE_PROPERTY "ro.product.device"
+#define PRODUCT_DEVICE_HDMIDONGLE     "hdmidongle"
 
 #include <inttypes.h>
 #include <math.h>
@@ -1556,6 +1558,10 @@ AudioPolicyManagerBase::AudioPolicyManagerBase(AudioPolicyClientInterface *clien
     mA2dpSuspended(false), mHasA2dp(false), mHasUsb(false), mHasRemoteSubmix(false),
     mSpeakerDrcEnabled(false)
 {
+    char property[PROPERTY_VALUE_MAX];
+    property_get(PRODUCT_DEVICE_PROPERTY, property, "");
+    mDeviceIsHdmidongle = (strstr(property, PRODUCT_DEVICE_HDMIDONGLE) != NULL);
+
     mpClientInterface = clientInterface;
 
     for (int i = 0; i < AudioSystem::NUM_FORCE_USE; i++) {
@@ -3254,7 +3260,7 @@ float AudioPolicyManagerBase::computeVolume(int stream,
     // if volume is not 0 (not muted), force media volume to max on digital output
     if (stream == AudioSystem::MUSIC &&
         index != mStreams[stream].mIndexMin &&
-        (device == AUDIO_DEVICE_OUT_AUX_DIGITAL ||
+        (((device == AUDIO_DEVICE_OUT_AUX_DIGITAL) && (mDeviceIsHdmidongle == false)) ||
          device == AUDIO_DEVICE_OUT_DGTL_DOCK_HEADSET)) {
         return 1.0;
     }
diff --git a/include/hardware_legacy/AudioPolicyManagerBase.h b/include/hardware_legacy/AudioPolicyManagerBase.h
index cf03f78..f33f25d 100644
--- a/include/hardware_legacy/AudioPolicyManagerBase.h
+++ b/include/hardware_legacy/AudioPolicyManagerBase.h
@@ -1,5 +1,6 @@
 /*
  * Copyright (C) 2009 The Android Open Source Project
+ * Copyright (C) 2013 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -572,6 +573,7 @@ protected:
                                 // to boost soft sounds, used to adjust volume curves accordingly
 
         Vector <HwModule *> mHwModules;
+        bool mDeviceIsHdmidongle;
 
 #ifdef AUDIO_POLICY_TEST
         Mutex   mLock;
-- 
1.8.0

